### DISCUSSION TITLE
Multiple teams/projects -  Multi Tenancy.

### DISCUSSION THREAD

**Discussion by stefanondisponibile** [argoproj/argo-workflows#6904] [Ideas]
Hello!
I'm wondering what's the best (or suggested) way to manage multiple teams, with the ability for each team to handle multiple projects.

For a comparison, in Azkaban this is quite easy:

![azkaban-project](https://azkaban.readthedocs.io/en/latest/_images/emptyprojectpage.png)

In Airflow there are you can filter by *tag*.
![airflow-tags](https://airflow.apache.org/docs/apache-airflow/stable/_images/dags.png)

**ArgoWorkflows** let us filter by *label*, but just for submitted workflows (*not* for WorkflowTemplates and CronWorkflows):

![argo-wf-1](https://user-images.githubusercontent.com/20441825/136701984-12858fda-9023-4da1-9804-e5028e20386d.png)
![argo-wf-2](https://user-images.githubusercontent.com/20441825/136701985-e7cb2ad0-a05f-4cf1-9118-74bc947d4e73.png)
![argo-wf-3](https://user-images.githubusercontent.com/20441825/136701987-b0a1e13a-d024-4073-91a1-8e60b4f0cc50.png)

What would you think about having the same filters we have in the **Workflows** section also in the other ones?
How do you currently handle multi tenancy?

---

Related:
* [Multi Tenancy for Development Teams](https://github.com/argoproj/argo-workflows/discussions/6337)
* [Multiple managed-namespace](https://github.com/argoproj/argo-workflows/issues/6769)
* [Multiple managed-namespace (**ArgoEvents**)](https://github.com/argoproj/argo-events/issues/1353)

**Comment by agilgur5** [comment]
> **ArgoWorkflows** let us filter by _label_, but just for submitted workflows (_not_ for WorkflowTemplates and CronWorkflows):

Workflow Templates have had a label filter since #6955, which was released in 3.3.
Cron Workflows have had a label filter since #7035, which was also released in 3.3.

So I believe this is now doable in current versions.

See also #11941 and #9312 for some "group" feature requests _on top of_ label and namespace filters.
